src/services/dom.d.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) Garuda Labs, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import type {
+  BeforeAfterParseHandler,
+  Fetch,
+  XResponseStaleReason,
+} from './types';
+import type { Document } from 'hyperview/src/types-legacy';
+
+/**
+ * TS definition of the Parser class to provide an alternate Document response
+ */
+export class Parser {
+  constructor(
+    fetch: Fetch,
+    onBeforeParse: ?BeforeAfterParseHandler,
+    onAfterParse: ?BeforeAfterParseHandler,
```

Comment:
are these the right way to define optional params? I thought the leading `?` was flow-type specific.

Reply:
You'd have to ask the author of that particular file ;).

I'll change it.

Reply:
Fixed.